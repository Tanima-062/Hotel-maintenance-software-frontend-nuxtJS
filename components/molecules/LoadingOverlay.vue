<template>
  <transition name="loading">
    <div v-if="isLoading" class="loading">
      <div class="loading-body">
        <Loading />
      </div>
    </div>
  </transition>
</template>
<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'

@Component({
  components: {
    Loading: () => import('~/components/atoms/Loading.vue'),
  },
})
export default class extends Vue {
  private isLoading: boolean = false

  private start(): void {
    this.isLoading = true
  }

  private finish(): void {
    this.isLoading = false
  }
}
</script>
<style lang="sass" scoped>
.loading
  filter: contrast(1)
  font-size: 30px
  position: fixed
  z-index: 1000
  top: 0
  right: 0
  bottom: 0
  left: 0
  text-align: center
  background: rgba(255, 255, 255, .4)

.loading-body
  position: absolute
  top: 50%
  left: 50%
  width: 40px
  height: 40px
  transform: translate(-50%, -50%)
  border-radius: 8px
</style>
