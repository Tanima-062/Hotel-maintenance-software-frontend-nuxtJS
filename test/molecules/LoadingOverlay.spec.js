import { mount } from '@vue/test-utils'
import LoadingOverlay from '@/components/molecules/LoadingOverlay.vue'
import Loading from '@/components/atoms/Loading.vue'

describe('Molecules LoadingOverlay Check', () => {
  test('LoadingOverlay Test', () => {
    const wrapper = mount(LoadingOverlay, {
      stubs: {
        Loading,
      },
    })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
