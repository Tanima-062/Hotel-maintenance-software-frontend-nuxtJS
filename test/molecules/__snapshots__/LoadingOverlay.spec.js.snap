// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules LoadingOverlay Check LoadingOverlay Test 1`] = `
<transition-stub
  name="loading"
>
  <!---->
</transition-stub>
`;
